testsuite/smoke-test/src/nft_transaction.rs

```diff
@@ -19,92 +14,99 @@ impl Test for NFTTransaction {
-        client.submit_and_wait(&receiver_register_txn).await?;
-
-        // transfer 42 tokens to receiver
-        let transfer_amount = 42;
-
-        // prepare transfer transaction
-        let test_sequence_number = client
-            .get_account(sender_address)
-            .await?
-            .into_inner()
-            .sequence_number;
-
-        let unsigned_txn = ctx
-            .transaction_factory()
-            .payload(
-                experimental_stdlib::encode_transfer_token_between_galleries_script_function(
-                    tokens::bars_tag(),
-                    receiver_address,
-                    transfer_amount,
-                    sender_address,
-                    0,
-                ),
+        let mut creator = ctx.random_account();
+        ctx.create_user_account(creator.public_key()).await?;
+        ctx.mint(creator.address(), 10_000_000).await?;
+
+        let mut owner = ctx.random_account();
+        ctx.create_user_account(owner.public_key()).await?;
+        ctx.mint(owner.address(), 10_000_000).await?;
+
+        let collection_builder = ctx.transaction_factory().payload(
+            aptos_stdlib::encode_create_unlimited_collection_script_script_function(
+                hex::encode("description").into_bytes(),
+                hex::encode("collection name").into_bytes(),
+                hex::encode("uri").into_bytes(),
+            ),
+        );
+
+        let collection_txn = creator.sign_with_transaction_builder(collection_builder);
+        client.submit_and_wait(&collection_txn).await?;
+
+        let token_builder = ctx.transaction_factory().payload(
+            aptos_stdlib::encode_create_token_script_script_function(
+                hex::encode("collection name").into_bytes(),
+                hex::encode("description").into_bytes(),
+                hex::encode("token name").into_bytes(),
+                1,
+                hex::encode("uri").into_bytes(),
+            ),
+        );
+
+        let token_txn = creator.sign_with_transaction_builder(token_builder);
+        client.submit_and_wait(&token_txn).await?;
+
+        let token_num = client
+            .get_account_resources_by_type(
+                creator.address(),
+                AccountAddress::from_hex_literal("0x1").unwrap(),
+                &"Token".parse().unwrap(),
+                &"Collections".parse().unwrap(),
             )
-            .sender(sender_address)
-            .sequence_number(test_sequence_number)
-            .build();
-        assert_eq!(unsigned_txn.sender(), sender_address);
-
-        // sign the transaction with the private key
-        let signature = sender_private_key.sign(&unsigned_txn);
-        // submit the transaction
-        let txn = SignedTransaction::new(unsigned_txn, sender_public_key, signature);
-        client.submit_and_wait(&txn).await?;
+            .await?
+            .inner()[0]
+            .data
+            .as_object()
```

Comment:
Wow... Probably not something we need to fix in this PR, but there's gotta be some better short hand to avoid this 😨